libs/base/Control/Monad/Maybe.idr

```diff
@@ -0,0 +1,39 @@
+module Control.Monad.Maybe
+
+import Control.Monad.Trans
+
+%access public export
+
+data MaybeT : (m : Type -> Type) -> (a : Type) -> Type where
+    MkMaybeT : m (Maybe a) -> MaybeT m a
+
+runMaybeT : MaybeT m a -> m (Maybe a)
+runMaybeT (MkMaybeT v) = v
+
+
+Functor m => Functor (MaybeT m) where
+    map f (MkMaybeT v) = MkMaybeT $ map (map f) v
+
+Applicative m => Applicative (MaybeT m) where
+    pure x = MkMaybeT (pure $ pure x)
+    (<*>) (MkMaybeT x) (MkMaybeT y) =
```

Comment:
You actually don't need `apply` and `both`, this line can be written as 
`(MkMaybeT x) <*> (MkMaybeT y) = MkMaybeT [| x <*> y |] `